packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -350,24 +350,26 @@ export interface SdkUnionType<TValueType extends SdkTypeBase = SdkType> extends
 
 export type AccessFlags = "internal" | "public";
 
-export interface SdkModelType extends SdkTypeBase {
+export interface SdkModelType<TPropertyType extends SdkModelPropertyType = SdkModelPropertyType>
+  extends SdkTypeBase {
   kind: "model";
-  properties: SdkModelPropertyType[];
+  properties: TPropertyType[];
   name: string;
   isGeneratedName: boolean;
   access: AccessFlags;
   usage: UsageFlags;
   additionalProperties?: SdkType;
   discriminatorValue?: string;
   discriminatedSubtypes?: Record<string, SdkModelType>;
-  discriminatorProperty?: SdkModelPropertyType;
+  discriminatorProperty?: TPropertyType;
   baseModel?: SdkModelType;
   crossLanguageDefinitionId: string;
   apiVersions: string[];
 }
 
-export interface SdkInitializationType extends SdkModelType {
-  properties: SdkParameter[];
+export interface SdkInitializationType {
+  access: AccessFlags;
```

Comment:
i've realized we can't modify the `.access` on our existing `SdkInitializationType` to denote whether a client is publicly or internally initialized. There is both the context that the client options model is used, and the client options model itself. In this case, the context that the client options model is used has `access` that depends on whether it's a client or og. However, that doesn't change the actual `access` of the client options model itself.

This is a breaking change, so want to call this out and will call it out more in our sync